src/output/visualizer/visualizer.py

```diff
@@ -12,6 +12,9 @@ def __init__(self, name):
         """
         self.name = name
 
+    def get_name(self):
```

Comment:
there's actually a nicer/cleaner way to do this in python:
* rename `self.name` to `self._name` (we should've anyway used that name; something i can clean up in a follow-up PR in all files)
* name this method here `name` and annotate it with `@property`. this will then allow you to use it as `visualizer.name` (instead of `visualizer.get_name()`), but it's a read-only property access instead of having read-write access to the member variable (as you'd get if you'd write `visualiser.name` right now which would also work since python doesn't have a visibility (private/public) concept)

see e.g. [this guide](https://www.geeksforgeeks.org/python-property-decorator-property/) on how `@property` works (note: here we don't need/want the setter & deleter which they also show - just the classic `@property`)